File: sdk/dart/test/put_constraint_request_test.dart
import 'package:test/test.dart';
import 'package:flagent_client/flagent_client.dart';

// tests for PutConstraintRequest
void main() {
  final instance = PutConstraintRequest((b) => b
    ..property = 'region'
    ..operator_ = 'IN'
    ..value = 'EU,US');

  group(PutConstraintRequest, () {
    test('to test the property `property`', () async {
      expect(instance.property, equals('region'));
    });

    test('to test the property `operator_`', () async {
      expect(instance.operator_, equals('IN'));
    });

    test('to test the property `value`', () async {
      expect(instance.value, equals('EU,US'));
    });
  });
}

File: sdk/java/src/test/java/com/flagent/client/model/PutConstraintRequestTest.java
package com.flagent.client.model;

import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;

/**
 * Model tests for PutConstraintRequest
 */
class PutConstraintRequestTest {

    @Test
    void testPutConstraintRequest() {
        PutConstraintRequest model = new PutConstraintRequest()
            .property("region")
            .operator("EQ")
            .value("EU");
        Assertions.assertNotNull(model);
        Assertions.assertEquals("region", model.getProperty());
        Assertions.assertEquals("EQ", model.getOperator());
        Assertions.assertEquals("EU", model.getValue());
    }

    @Test
    void propertyTest() {
        PutConstraintRequest model = new PutConstraintRequest().property("tier");
        Assertions.assertEquals("tier", model.getProperty());
    }

    @Test
    void operatorTest() {
        PutConstraintRequest model = new PutConstraintRequest().operator("NEQ");
        Assertions.assertEquals("NEQ", model.getOperator());
    }

    @Test
    void valueTest() {
        PutConstraintRequest model = new PutConstraintRequest().value("free");
        Assertions.assertEquals("free", model.getValue());
    }

}
